cmd/template/main.go

```diff
@@ -23,6 +23,50 @@ func main() {
 `)
 }
 
+// WebsocketTemplate returns a slice that represents the logic of the websocket
+// handler. This function expects some variables set in the template. See echo
+// router for more details.
+//
+// w http.ResponseWriter
+// r *http.Request
+func websocketTemplate() []byte {
+	return []byte(`
+	errorMessage := []byte("This is another message not PING")
+	socket, err := websocket.Accept(w, r, nil)
+	defer socket.Close(websocket.StatusGoingAway, "server closing websocket")
```

Comment:
You'll want to check the error first, then defer the close. Otherwise you will get what we saw a stream. A Panic because we tried to close a nil socket.